File: 2022/day1/cpp/part2.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <vector>

using namespace std;

int main() {
    ifstream f("input.txt");
    string s;
    int sum{};
    vector<int> elves;
    while (getline(f, s)) {
        if (s != "") {
            sum += stoi(s);
        } else {
            elves.push_back(sum);
            sum = 0;
        }
    }
    sort(elves.rbegin(), elves.rend());
    cout << elves[0] + elves[1] + elves[2] << endl;
    f.close();
}

File: 2022/day1/elixir/part2.exs
{:ok, file} = File.open("../input.txt", [:read])

contents = IO.read(file, :all) |> String.split("\n\n")

lists =
  Enum.map(contents, fn section ->
    section
    |> String.split("\n")
    |> Enum.map(&String.to_integer/1)
    |> Enum.sum()
  end)

Enum.sort(lists, :desc) |> Enum.take(3) |> Enum.sum() |> IO.puts()

File.close(file)
